openapi_python_client/parser/responses.py

```diff
@@ -79,27 +80,30 @@ def empty_response(
     )
 
 
-def response_from_data(
+def response_from_data(  # noqa: PLR0911
     *,
     status_code: HTTPStatus,
     data: Union[oai.Response, oai.Reference],
     schemas: Schemas,
+    responses: Dict[str, Union[oai.Response, oai.Reference]],
     parent_name: str,
     config: Config,
 ) -> Tuple[Union[Response, ParseError], Schemas]:
     """Generate a Response from the OpenAPI dictionary representation of it"""
 
     response_name = f"response_{status_code}"
     if isinstance(data, oai.Reference):
-        return (
-            empty_response(
-                status_code=status_code,
-                response_name=response_name,
-                config=config,
-                data=data,
-            ),
-            schemas,
-        )
+        ref_path = parse_reference_path(data.ref)
+        if isinstance(ref_path, ParseError):
+            return ref_path, schemas
+        if not ref_path.startswith("/components/responses/"):
+            return ParseError(data=data, detail=f"$ref to {data.ref} not allowed in responses"), schemas
+        resp_data = responses.get(ref_path.split("/")[-1], None)
+        if not resp_data:
+            return ParseError(data=data, detail=f"Could not find reference: {data.ref}"), schemas
+        if not isinstance(resp_data, oai.Response):
+            return ParseError(data=data, detail="Top-level $ref inside components/responses is not supported"), schemas
```

Comment:
instead of individual return statements like these,... what if we used a `error_detail: str` variable and just have one return statement at the end of this function that has one `return` statement for `ParseError` -- this may or may not look very pretty but at least it would avoid disabling `PLR0911` rule

Reply:
I honestly find it easier to follow the logic when there's explicit short-circuiting like this. Maybe it's because I did a lot of Go and that's the preferred pattern in Go. But if I see a bunch of this kind of thing—

```
if bad_condition_1:
    error_detail = message1
elif bad_condition_2:
    error_detail = message2
else:
    error_detail = None
if error_detail:
    return ParseError(data=data, detail=error_detail)
```

—it bothers me a little because now the code paths for the error conditions are not really independent of each other; if for instance I messed up and put "if" instead of "elif" for the second one, now the bad_condition_1 test would be effectively a no-op. It also means that I have to be constructing the same kind of error, ParseError, each time (and the same way each time: `data=data`), which happens to be true in this case but isn't logically inevitable; if I had to make a change where now one of those cases raises a different error, I'd have to refactor the whole thing.

PLR0911 is only a helpful rule when it's helpful, and this codebase disables it in quite a few parsing/validation functions like this.

Reply:
>PLR0911 is only a helpful rule when it's helpful, and 
agree

> this codebase disables it in quite a few parsing/validation functions like this

okay this sounds good to me